odl/operator/fn_ops.py

```diff
@@ -0,0 +1,406 @@
+        return self.__flat_indices
+
+    @property
+    def flat_indices_unique(self):
+        """unique set of indices where to sample the function."""
+        return self.__flat_indices_unique
+
+    def _call(self, x, out=None):
+        """sum all values if indices are given multiple times."""        
+        y = np.bincount(self.flat_indices, weights=x, minlength=self.range.size)
+            
+        if out is None:
+            out = self.range.element(y)
+        else:
+            out[:] = y
+
+        return out
+
+    @property
+    def adjoint(self):
+        """Adjoint of the sampling is 'sampling'.
+
+        Returns
+        -------
+        adjoint : `SamplingOperator`
+        """
+        return SamplingOperator(self.domain, self.indices)
+
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        return '{}({!r}, {!r})'.format(self.__class__.__name__,
+                                       self.domain, self.indices)
+                
+
+    def __str__(self):
+        """Return ``str(self)``."""
+        return '{} * I'.format(self.indices)
+
+
+class VectoriseOperator(Operator):
+
+    """Operator that reshapes the object as a column vector.
+
+        ``VectoriseOperator(f) == f[:]``
+    """
+
+    def __init__(self, domain, scaling=1):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        domain : `LinearSpace` or `Field`
+            Set of elements on which this operator acts.
+
+        Examples
+        --------
+        >>> X = odl.uniform_discr(min_pt=[-1, -1], max_pt=[1, 1], shape=[3, 4])
+        >>> x = X.element(range(X.size))
+        >>> A = odl.operator.VectoriseOperator(X)
+        >>> A(x)
+        rn(12).element([0.0, 1.0, 2.0, ..., 9.0, 10.0, 11.0])
+        >>> A(x)  # Out-of-place
+        rn(12).element([0.0, 1.0, 2.0, ..., 9.0, 10.0, 11.0])
+
+        >>> X = odl.uniform_discr(min_pt=[-1, -1], max_pt=[1, 1], shape=[3, 4])
+        >>> x = X.element(range(X.size))
+        >>> A = odl.operator.VectoriseOperator(X)
+        >>> (A.inverse(A(x)) - x).norm()
+        0.0
+        >>> (A.inverse(A(x)) - x).norm() # Out-of-place
+        0.0
+        """
+        if not isinstance(domain, (LinearSpace, Field)):
+            raise TypeError('`space` {!r} not a `LinearSpace` or `Field` '
+                            'instance'.format(domain))
+        
+        self.__scaling = scaling        
+        range = fn(domain.size, dtype=domain.dtype)
+        super().__init__(domain, range, linear=True)
+
+    def _call(self, x, out=None):
+        """Collect indices."""
+        if out is None:
+            out = self.scaling*x[:]
+        else:
+            out[:] = self.scaling*x[:]
+        return out
+
+    @property
+    def scaling(self):
+        """scaling factor."""
+        return self.__scaling
+    
+    @property
+    def adjoint(self):
+        """Adjoint of the vectorising is creating an element with these components
```

Comment:
Thanks. I wasn't aware of that spyder feature but I like it a lot! It makes life so much easier.